from_now/from_now.go

```diff
@@ -0,0 +1,112 @@
+package from_now
+
+import (
+	"fmt"
+	"github.com/taskcluster/taskcluster-cli/extpoints"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type from_now struct{}
+
+func init() {
+	extpoints.Register("from-now", from_now{})
+}
+
+func (from_now) ConfigOptions() map[string]extpoints.ConfigOption {
+	return nil
+}
+
+func (from_now) Summary() string {
+	return "Returns a timestamp which is DURATION ahead in the future."
+}
+
+func (from_now) Usage() string {
+	usage := "Usage: taskcluster from-now DURATION"
+	usage += "\n"
+	return usage
+}
+
+func (from_now) Execute(context extpoints.Context) bool {
+	duration := context.Arguments["DURATION"].(string)
+
+	offset, err := parseTime(duration)
+	
+	if err {
+		panic("String: '" + duration + "' isn't a time expression")
+	}
+
+	timeToAdd := time.Hour*time.Duration(offset.weeks*7*24) +
+		time.Hour*time.Duration(offset.days*24) +
+		time.Hour*time.Duration(offset.hours) +
+		time.Minute*time.Duration(offset.minutes) +
+		time.Second*time.Duration(offset.seconds)
+
+	timein := time.Now().Add(timeToAdd)
+	timein = timein.AddDate(offset.years, offset.months, 0)
+
+	fmt.Println(timein.Format(time.RFC3339))
+
+	return true
+}
+
+type parse_time struct {
+	years   int
+	months  int
+	weeks   int
+	days    int
+	hours   int
+	minutes int
+	seconds int
+}
+
+/* parseTime takes an argument `str` which is a string on the form `1 day 2 hours 3 minutes`
+ * where specification of day, hours and minutes is optional. You can also the
+ * short hand `1d2h3min`, it's fairly tolerant of different spelling forms and
+ * whitespace. But only really meant to be used with constants.
+ *
+ * Returns a parse_time object with all of the fields filled in with the correct values.
+ */
+func parseTime(str string) (parse_time, bool) {
+
+	// Regexp taken from github.com/taskcluster/taskcluster-client/blob/master/lib/parsetime.js
+	reg := []string{
+		"^(\\s*(-|\\+))?",
+		"(\\s*(\\d+)\\s*y((ears?)|r)?)?",
+		"(\\s*(\\d+)\\s*mo(nths?)?)?",
+		"(\\s*(\\d+)\\s*w((eeks?)|k)?)?",
+		"(\\s*(\\d+)\\s*d(ays?)?)?",
+		"(\\s*(\\d+)\\s*h((ours?)|r)?)?",
+		"(\\s*(\\d+)\\s*min(utes?)?)?",
+		"(\\s*(\\d+)\\s*s(ec(onds?)?)?)?",
+		"\\s*$",
+	}
+
+	re := regexp.MustCompile(strings.Join(reg, ""))
+
+	if !re.MatchString(str) {
+		return parse_time{}, true
+	}
+
+	groupMatches := re.FindAllStringSubmatch(str, -1)
+
+	offset := parse_time{}
+
+	// Add negative support after we figure out what we are doing with docopt because it complains about the '-'
```

Comment:
I think once waiting to figure out what we're doing with _docopt_ makes sense here because if we scrap it adding the minus command would be useless.